tests/core/test_tracker_stores.py

```diff
@@ -388,10 +407,84 @@ def test_set_fail_safe_tracker_store_domain(default_domain: Domain):
     assert fallback_tracker_store.domain is failsafe_store.domain
 
 
-def test_sql_tracker_store_retrieve_with_session_started_events(default_domain: Domain):
-    tracker_store = SQLTrackerStore(default_domain, host="sqlite:///")
+def create_tracker_with_partially_saved_events(
+    tracker_store: TrackerStore,
+) -> Tuple[List[Event], DialogueStateTracker]:
+    # creates a tracker with two events and saved it to the tracker store
+    # following that, it adds three more events that are not saved to the tracker store
+    sender_id = uuid.uuid4().hex
 
-    # Create tracker with a SessionStarted event
+    # create tracker with two events and save it
+    events = [UserUttered("hello"), BotUttered("what")]
+    tracker = DialogueStateTracker.from_events(sender_id, events)
+    tracker_store.save(tracker)
+
+    # add more events to the tracker, do not yet save it
+    events = [
+        ActionExecuted(ACTION_LISTEN_NAME),
+        UserUttered("123"),
+        BotUttered("yes"),
+    ]
+    for event in events:
+        tracker.update(event)
+
+    return events, tracker
+
+
+def test_mongo_additional_events(default_domain: Domain):
+    tracker_store = MockedMongoTrackerStore(default_domain)
+    events, tracker = create_tracker_with_partially_saved_events(tracker_store)
+
+    # make sure only new events are returned
+    # noinspection PyProtectedMember
+    assert list(tracker_store._additional_events(tracker)) == events
+
+
+# we cannot parametrise over this and the previous test due to the different ways of
+# calling _additional_events()
+def test_sql_additional_events(default_domain: Domain):
+    tracker_store = SQLTrackerStore(default_domain)
+    events, tracker = create_tracker_with_partially_saved_events(tracker_store)
+
+    # make sure only new events are returned
+    with tracker_store.session_scope() as session:
+        # noinspection PyProtectedMember
+        assert list(tracker_store._additional_events(session, tracker)) == events
+
+
+def test_mongo_events_since_last_session_start(default_domain: Domain):
```

Comment:
isn't that test covered by the `parametrized` test?

Reply:
it is implicitly - do you think it should not be repeated explicitly?

Reply:
is there a case where one would fail, but the other one wouldn't? If not, then I'd remove this one.

Reply:
no I don't think there's such a case, will remove it